plugins/module_utils/network/iosxr/iosxr.py

```diff
@@ -575,7 +575,34 @@ def load_config(
             diff = response.get("show_commit_config_diff")
 
         except ConnectionError as exc:
-            module.fail_json(msg=to_text(exc, errors="surrogate_then_replace"))
+            try:
+                if "Invalid input detected" in str(exc) and "comment" in str(
+                    exc
+                ):
+                    comment = None
+                    response = conn.edit_config(
+                        candidate=command_filter,
```

Comment:
This will issue the config statement twice if in case the commit comment is not supported. How about handling it in the cliconf plugin itself https://github.com/ansible-collections/cisco.iosxr/blob/main/plugins/cliconf/iosxr.py#L180, if a commit with comment fails send commit with a comment as `None`.

Reply:
I tried that but when I added try except in cliconf the except block is not able to caught the exception and the control is coming back to load_config function

Reply:
let me try this one more time

Reply:
Try catching `AnsibleConnectionFailure` exception in cliconf plugin after commit

Reply:
ack